src/modules/bootloader/main.py

```diff
@@ -637,42 +637,61 @@ def run_grub_install(fw_type, partitions, efi_directory):
                                    "--bootloader-id=" + efi_bootloader_id,
                                    "--force"])
     else:
-        assert efi_directory is None
-        if libcalamares.globalstorage.value("bootLoader") is None:
-            return
+        if efi_directory is not None:
+            if not install_hybrid_grub:
+                return
 
-        boot_loader = libcalamares.globalstorage.value("bootLoader")
-        if boot_loader["installPath"] is None:
-            return
+        if libcalamares.globalstorage.value("bootLoader") is None:
+            efi_install_path = libcalamares.globalstorage.value("efiSystemPartition")
+            if efi_install_path is None or efi_install_path == "":
+                efi_install_path = "/boot/efi"
+            find_esp_disk_command = f"echo \"/dev/$(lsblk -o PKNAME $(df --output=source '{efi_install_path}' | tail -n1) | tail -n1)\""
```

Comment:
I get the feeling we're doing things in shell that we don't have to. `lsblk $(df | tail -n1)` ok, you want the substitution there, and *maybe* the second `tail -n1` as well, but why do string-manipulation there as well, .. you could do this in one or more steps with target_env_output and then add the `/dev/` bit in Python.